src/internal/obj/microsoft_client.go

```diff
@@ -39,16 +39,30 @@ func newMicrosoftClient(container string, accountName string, accountKey string)
 	return &microsoftClient{container: (&blobSvc).GetContainerReference(container)}, nil
 }
 
-func (c *microsoftClient) Writer(ctx context.Context, name string) (io.WriteCloser, error) {
-	return newMicrosoftWriter(ctx, c, name), nil
+func (c *microsoftClient) Put(ctx context.Context, name string, r io.Reader) error {
+	w := newMicrosoftWriter(ctx, c, name)
+	if _, err := io.Copy(w, r); err != nil {
+		w.Close()
```

Comment:
A TODO to remove this and respect the context would probably be good here. Since this operation is idempotent, closing a partial write and returning the error should be fine for now.